src/lib/coreboot_table.c

```diff
@@ -388,6 +390,50 @@ void __weak lb_board(struct lb_header *header) { /* NOOP */ }
  */
 void __weak lb_spi_flash(struct lb_header *header) { /* NOOP */ }
 
+
+/*
+ * Loads the logo bitmap file from the BOOTSPLASH fmap region into CBMEM
+ */
+static void tianocore_logo_load(void)
+{
+	const struct cbmem_entry *logo_entry;
+	struct region_device rdev;
+	struct bootlogo_header header;
+	void *logo_buffer;
+
+	struct cbfsf logo_file;
+	uint32_t compression;
+	uint32_t cbfs_type = CBFS_TYPE_RAW;
+	size_t logo_size;
+
+	logo_entry = cbmem_entry_add(CBMEM_ID_TIANOCORE_LOGO, 4 * MiB);
```

Comment:
Why don't we create the memory buffer only as big as the logo + header with size? It is a waste of memory

Reply:
The buffer in cbmem is larger to accomodate uncompressed bitmaps. The file in CBFS may be up to 1MB compressed and up to 4 MB uncompressed.

Reply:
I understand it pretty clearly, but we can create cbmem entry after the bitmap is loaded and uncompressed, note before. That way we may alloc just the amount of memory we need.

Reply:
Right. Updated as per suggestion